cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/TemplatesJarUtil.java

```diff
@@ -95,6 +101,66 @@ public static String downloadLatestDevon4jTemplates(boolean isDownloadSource, Fi
         TemplatesJarConstants.DEVON4J_TEMPLATES_ARTIFACTID, "LATEST", isDownloadSource, templatesDirectory);
   }
 
+  /**
+   * Downloads multiple jar files defined by the maven coordinates. Only downloads if files are not present or adapted
+   * folder does not exist.
+   *
+   * @param templatesDirectory directory where the templates jar are located
+   * @param mavenCoordinates list with {@link MavenCoordinate} that will be loaded
+   */
+  public static void downloadTemplatesByMavenCoordinates(File templatesDirectory,
+      List<MavenCoordinate> mavenCoordinates) {
+
+    if (mavenCoordinates == null || mavenCoordinates.isEmpty()) {
+      return;
+      // no templates specified
+    }
+    boolean downloadedExists = false;
+    for (File f : templatesDirectory.listFiles()) {
+      if (f.getName().equals(ConfigurationConstants.ADAPTED_FOLDER)) {
+        LOG.debug("Found adapted folder no download of templates needed");
+        return;
+      } else if (f.getName().equals(ConfigurationConstants.DOWNLOADED_FOLDER)) {
+        downloadedExists = true;
+      }
+    }
+    File downloadedDirectory = templatesDirectory.toPath().resolve(ConfigurationConstants.DOWNLOADED_FOLDER).toFile();
+    if (!downloadedExists) {
+      LOG.info("downloaded folder could not be found and will be created ");
+      try {
+        Files.createDirectory(templatesDirectory.toPath().resolve(ConfigurationConstants.DOWNLOADED_FOLDER));
+      } catch (IOException e) {
+        throw new CobiGenRuntimeException("Could not create Download Folder", e);
+      }
+    }
+
+    Set<MavenCoordinate> afterExistanceCheckList = new HashSet();
```

Comment:
Exist**E**nce
I think the variable name could be a bit better as well, if you want to check if the templates exist, why not simply "checkExistenceList"